### PR TITLE
Fix not using the users configured shell for sync

### LINKED ISSUE #5976: Can't connect to clusters (with OIDC)
**Describe the bug**

I am trying to connect to my company's Kubernetes clusters from my work laptop. All works fine via command line (`kubectl ...`) and via `k9s`, but I cannot get Lens to work.

When I try to connect to any of my configured clusters, I get the errors I describe below:
* either the `kubectl` executable is not found
* or some plugins are not found (`oidc-login`)

I also tried to specify the path of `kubectl` explicitly in Settings: `/usr/bin/kubectl`, but that made no difference.

I am not sure what exactly the problem could be, but it almost looks like something related to loading the shell profile is a bit broken, because, as I mentioned, everything works via Command Line (and k9s launched via command line). 

Also, I see this at the start of my logs:
```
error:   ▪ shellEnv: Error: Resolving shell environment is taking very long. Please review your shell configuration.
```

**To Reproduce**
I am not sure how to make this reproducible to someone else, but for me, it happens every single time I try to connect to a cluster.

**Expected behavior**
I would expect to be able to connect to the cluster fine, given that:
* I can user `kubectl` via CLI and it works perfectly fine
* I can use `k9s` and it works perfectly fine

**Environment (please complete the following information):**
- Lens Version: latest 6.0.0-latest.20220728.2, but it was happening in all 5.x I tried
- OS: Ubuntu 20.04 
- Installation method: downloaded the `.deb` from the webpage
- I use ZSH (with Oh-My-Zsh)
- I tried to execute the `lens` executable both from command line (both from `bash` or `zsh`) and from the window manager

**Logs:**
This is the error I can see in the log:
```
error:   ▪ [CLUSTER]: Failed to connect to "citi": StatusCodeError: 500 - "getting credentials: exec: executable kubectl not found\n\nIt looks like you are trying to use a client-go credential plugin that is not installed.\n\nTo learn more about this feature, consult the documentation available at:\n      https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins\n"
```

**Kubeconfig:**
```
- name: cel-1
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      args:
      - oidc-login
      - get-token
      - --oidc-issuer-url=....
      - --oidc-client-id=cel-1
      - --oidc-client-secret=.....
      - --oidc-extra-scope=email
      - --oidc-extra-scope=groups
      - --listen-address=127.0.0.1:18000
      command: kubectl
      env: []
      interactiveMode: IfAvailable
      provideClusterInfo: false
```

**Additional context**
I also tried to change the `kubeconfig` to use `command: /usr/bin/kubectl`, and the error message change to:

```
error:   ▪ [CLUSTER]: Failed to connect to "develop": StatusCodeError: 500 - "getting credentials: exec: executable /usr/bin/kubectl failed with exit code 1\n" +71ms
```

And it looks like this in the UI:
error

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes https://github.com/lensapp/lens/issues/5976

### CODE REVIEW

**Review by jim-docker — COMMENTED** [review]
Node shell still doesn't use preferences value(?)

https://user-images.githubusercontent.com/40840436/202317893-4ad15a9a-5bca-42a8-813e-e7a625ebd562.mov

**Comment by Nokel81** [comment]
@jim-docker That is expected behaviour. Node shells are on different machines so the paths are not guaranteed to be the same.

File: src/common/user-store/user-store.ts [hunk 1]
```diff
@@ -91,10 +91,6 @@ export class UserStore extends BaseStore<UserStoreModel> /* implements UserStore
    */
   @observable syncKubeconfigEntries!: StoreType<typeof DESCRIPTORS["syncKubeconfigEntries"]>;
 
-  @computed get resolvedShell(): string | undefined {
-    return this.shell || process.env.SHELL || process.env.PTYSHELL;
```
Comment (jim-docker) [inline src/common/user-store/user-store.ts:95]: By removing this does it mean we no longer fall back on process.env values? Not necessary anymore?

Comment (Nokel81) [reply src/common/user-store/user-store.ts:95]: It is not necessary anymore since user information reads from the system

**Review by jim-docker — APPROVED** [review]
Definitely fixes a problem but not able to reproduce linked issue to confirm it has been fixed

**Pull request merged**